node/hyperbahn/service_proxy.js

```diff
@@ -166,7 +173,17 @@ function createServiceChannel(serviceName) {
         }
     }
 
-    svcchan.handler = new RelayHandler(svcchan);
+    var handler = new RelayHandler(svcchan);
+
+    // Decorate a circuit health monitor to egress request handlers.
+    if (mode === 'exit' && self.circuits) {
+        handler = new ServiceHealthProxy({
```

Comment:
do we want to go this proxy handler approach ?

We could just have `RelayHandler` itself hook directly into the circuits code. This would avoid a bunch of wrapping and closures.